frames/section.lua

```diff
@@ -59,6 +62,10 @@ function sectionProto:SetTitle(text)
   self.title:SetText(text)
 end
 
+function sectionProto:SetBagKind(bagkind)
```

Comment:
Change this to SetKind

Add annotations for the parameter

Use the property `kind` not `bagkind`

Add the field `kind` with the correct Enum type in the section prototype.

Reply:
Also, reset the kind on release.